__tests__/app.test.js

```diff
@@ -22,133 +20,174 @@ describe('api',()=>{
 
-  // a review object, which should have the following properties:
+          const foundReview = reviews.find((review) => {
+            if (review.review_id === 2) {
+              return review;
+            }
+          });
 
-  describe("/api/reviews/review_id", () => {
-    it("200 - responds with single review object", () => {
+          expect(foundReview.comment_count).toBe("3");
+        });
+    });
+    it("200 GET - the reviews should be sorted by date in descending order", () => {
       return request(app)
-        .get("/api/reviews/2")
+        .get("/api/reviews")
         .expect(200)
         .then(({ body }) => {
-          const review = body.review;
-          //Should be an Object
-
-          console.log(review);
-
-          expect(review.review_id).toBe(2);
-          expect(review.title).toBe('Jenga');
-          expect(review.category).toBe('dexterity');
-          expect(review.designer).toBe('Leslie Scott');
-          
-          expect(review.owner).toBe('philippaclaire9');
-          expect(review.review_body).toBe('Fiddly fun for all the family');
-          expect(review.review_img_url).toBe('https://images.pexels.com/photos/4473494/pexels-photo-4473494.jpeg?w=700&h=700');
-          expect(review.created_at).toBe('2021-01-18T10:01:41.251Z');
-          expect(review.votes).toBe(5);
+          const reviews = body.reviews;
+
+          const reviewDates = reviews.map((review) => {
+            return review.created_at;
+          });
+          expect(reviewDates).toBeSorted({ descending: true });
         });
     });
+  });
 
-    it("400 - responds with msg bad request", () => {
+  describe("/api/reviews/:review_id/comments", () => {
+    it("201 POST - responds with the posted comment", () => {
       return request(app)
-        .get("/api/reviews/bad-request")
-        .expect(400)
+        .post("/api/reviews/6/comments")
+        .send({ username: "mallionaire", body: "This is a new comment!" })
+        .expect(201)
+        .then(({ body }) => {
+
+          const comment = body.comment;
+
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            body: "This is a new comment!",
+            review_id: 6,
+            author: "mallionaire",
+            votes: 0,
+            created_at: expect.any(String),
+          });
+        });
+    });
+    it("404 POST - responds with msg when sent valid but non-existent path", () => {
+      return request(app)
+        .post("/api/reviews/99999999/comments")
+        .send({ username: "mallionaire", body: "This is a new comment!" })
+        .expect(404)
         .then(({ body }) => {
           const serverResponseMsg = body.msg;
-          expect(serverResponseMsg).toBe("Bad request");
+          expect(serverResponseMsg).toBe("Review not found.");
         });
     });
 
-    it("404 - responds with msg when sent valid but non-existent path", () => {
+    it("400 POST - responds with bad request msg.", () => {
```

Comment:
We also want to add another 400 for if there are missing fields (this will need to be dealt with via promise.reject) - for instance if we make a post with no username or body